ga4gh/backend.py

```diff
@@ -552,6 +552,48 @@ def variantAnnotationsGenerator(self, request):
             request, variantAnnotationSet)
         return intervalIterator
 
+    def featuresGenerator(self, request):
+        """
+        :param request: JSON string - the original web request
+        :return: an iterator over the result set/nextPageToken pairs
+        """
+        compoundId = None
+        parentId = None
+        if request.featureSetId is not None:
+            compoundId = datamodel.FeatureSetCompoundId.parse(
+                request.featureSetId)
+        if request.parentId is not None and request.parentId != "":
+            compoundParentId = datamodel.FeatureCompoundId.parse(
+                request.parentId)
+            parentId = compoundParentId.featureId
+            # A client can optionally specify JUST the (compound) parentID,
+            # and the server needs to derive the dataset & featureSet
+            # from this (compound) parentID.
+            if compoundId is None:
+                compoundId = compoundParentId
+            else:
+                # check that the dataset and featureSet of the parent
+                # compound ID is the same as that of the featureSetId
+                mismatchCheck = (compoundParentId.datasetId !=
```

Comment:
This will be more robust to code changes (think snake_case) if you use hanging indent, e.g.
```python
 mismatchCheck = (
    compoundParentId.datasetId != compoundId.datasetId or
    compoundParentId.featureSetId !=compoundId.featureSetId)
```